src/main/java/io/github/genomicdatainfrastructure/discovery/datasets/infrastructure/beacon/persistence/BeaconIndividualsRequestMapper.java

```diff
@@ -0,0 +1,125 @@
+
+        var requestParameters = nonNullFacets.stream()
+                .filter(it -> BEACON_FACET_GROUP.equals(it.getSource()))
+                .filter(it -> FilterType.ENTRIES.equals(it.getType()))
+                .map(BeaconIndividualsRequestMapper::buildBeaconRequestParameters)
+                .toList();
+
+        return BeaconIndividualsRequest.builder()
+                .meta(new BeaconIndividualsRequestMeta())
+                .query(BeaconIndividualsRequestQuery.builder()
+                        .includeResultsetResponses(INCLUDE_RESULTSET_RESPONSES)
+                        .requestedGranularity(REQUESTED_GRANULARITY)
+                        .testMode(false)
+                        .pagination(new BeaconIndividualsRequestQueryPagination())
+                        .filters(beaconFilters)
+                        .requestParameters(requestParameters)
+                        .build())
+                .build();
+    }
+
+    private BeaconIndividualsRequestQueryFilter buildBeaconFilter(DatasetSearchQueryFacet facet) {
+        return switch (facet.getType()) {
+            case DROPDOWN -> buildDropdownBeaconFacet(facet);
+            case FREE_TEXT -> buildFreeTextBeaconFacet(facet);
+            default -> null;
+        };
+    }
+
+    private BeaconIndividualsRequestQueryFilter buildDropdownBeaconFacet(
+            DatasetSearchQueryFacet facet) {
+        String value = ofNullable(facet.getValue())
+                .filter(v -> !v.isBlank())
+                .orElseThrow(() -> new BadRequestException(
+                        "Facet value must not be null or empty"));
+
+        return BeaconIndividualsRequestQueryFilter.builder()
+                .id(value)
+                .scope(SCOPE)
+                .build();
+    }
+
+    private BeaconIndividualsRequestQueryFilter buildFreeTextBeaconFacet(
+            DatasetSearchQueryFacet facet) {
+        String operator = ofNullable(facet.getOperator())
+                .map(it -> it.value())
+                .orElseThrow(() -> new BadRequestException(
+                        "Facet operator must not be null"));
+
+        String value = ofNullable(facet.getValue())
+                .filter(v -> !v.isBlank())
+                .orElseThrow(() -> new BadRequestException(
+                        "Facet value must not be null or empty"));
+
+        String key = ofNullable(facet.getKey())
+                .filter(k -> !k.isBlank())
+                .orElseThrow(() -> new BadRequestException("Facet key must not be null or empty"));
+
+        return BeaconIndividualsRequestQueryFilter
+                .builder()
+                .id(key)
+                .operator(operator)
+                .value(value)
+                .scope(SCOPE)
+                .build();
+    }
+
+    private Map<String, String> buildBeaconRequestParameters(DatasetSearchQueryFacet facet) {
+        var entries = ofNullable(facet
+                .getEntries())
+                .filter(e -> !e.isEmpty())
```

Comment:
suggestion: Here and all other references where you use `!` to negate in a filter, it would be more readable if you use `Predicates.not` instead.